graylog2-server/src/main/java/org/graylog/plugins/views/search/elasticsearch/searchtypes/pivot/buckets/ESDateRangeHandler.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.plugins.views.search.elasticsearch.searchtypes.pivot.buckets;
+
+import io.searchbox.core.SearchResult;
+import io.searchbox.core.search.aggregation.DateRangeAggregation;
+import org.elasticsearch.search.aggregations.AggregationBuilder;
+import org.elasticsearch.search.aggregations.AggregationBuilders;
+import org.elasticsearch.search.aggregations.bucket.range.date.DateRangeAggregationBuilder;
+import org.graylog.plugins.views.search.Query;
+import org.graylog.plugins.views.search.elasticsearch.ESGeneratedQueryContext;
+import org.graylog.plugins.views.search.elasticsearch.searchtypes.pivot.ESPivot;
+import org.graylog.plugins.views.search.elasticsearch.searchtypes.pivot.ESPivotBucketSpecHandler;
+import org.graylog.plugins.views.search.searchtypes.pivot.Pivot;
+import org.graylog.plugins.views.search.searchtypes.pivot.buckets.DateRangeBucket;
+import org.joda.time.base.AbstractDateTime;
+
+import javax.annotation.Nonnull;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class ESDateRangeHandler extends ESPivotBucketSpecHandler<DateRangeBucket, DateRangeAggregation> {
```

Comment:
We have some basic tests for the other bucket handler. How about creating one for this handler, too? :slightly_smiling_face:

Reply:
I'm not really sure that's worth the trouble, what is it supposed to test? There's barely any logic in there.
Besides, there is no test for `ESValuesHandler`  :-P

Reply:
Okay, fair enough. :smile: